io/disk/fiotest.py

```diff
@@ -44,21 +45,46 @@ def setUp(self):
         """
         Build 'fio'.
         """
-        tarball = self.fetch_asset('http://brick.kernel.dk/snaps/fio-2.1.10.tar.gz')
+        default_url = "http://brick.kernel.dk/snaps/fio-2.1.10.tar.gz"
+        self.url = self.params.get('fio_tool_url', default=default_url)
+        tarball = self.fetch_asset(self.url)
         archive.extract(tarball, self.srcdir)
         fio_version = os.path.basename(tarball.split('.tar.')[0])
         self.srcdir = os.path.join(self.srcdir, fio_version)
         build.make(self.srcdir)
+        self.disk = self.params.get('disk', default=None)
+        self.dir = self.params.get('dir', default=self.srcdir)
+        self.fstype = self.params.get('fs', default='ext4')
+
+        if self.disk is not None:
+            self.part_obj = Partition(self.disk, mountpoint=self.dir)
+            self.log.info("Unmounting disk/dir before creating file system")
+            self.part_obj.unmount()
+            self.log.info("creating file system")
+            self.part_obj.mkfs(self.fstype)
+            self.log.info("Mounting disk %s on directory %s",
+                          self.disk, self.dir)
+            self.part_obj.mount()
```

Comment:
@PraveenPenguin  you want me to move following lines above the  build.make(self.srcdir)
+        self.disk = self.params.get('disk', default=None)
+        self.dir = self.params.get('dir', default=self.srcdir)
+        self.fstype = self.params.get('fs', default='ext4')

or

+            self.part_obj.mkfs(self.fstype)
+            self.log.info("Mounting disk %s on directory %s",
+                          self.disk, self.dir)
+            self.part_obj.mount()
?

Because the lines you have pasted are operations which I feel it is better to have these operations after build.make(self.srcdir). and thinking instead of pasting line55-line57 you pasted
line64-line67
please correct me if I am wrong, so that I can make the changes correctly.